llvm/lib/CodeGen/MachineVerifier.cpp

```diff
@@ -2414,6 +2414,34 @@ void MachineVerifier::visitMachineInstrBefore(const MachineInstr *MI) {
     }
     break;
   }
+  case TargetOpcode::COPY_LANEMASK: {
+    const MachineOperand &DstOp = MI->getOperand(0);
+    const MachineOperand &SrcOp = MI->getOperand(1);
+    const MachineOperand &LaneMaskOp = MI->getOperand(2);
+    const Register SrcReg = SrcOp.getReg();
+    const Register DstReg = DstOp.getReg();
+    const LaneBitmask LaneMask = LaneMaskOp.getLaneMask();
+
+    if (!SrcReg.isPhysical() || !DstReg.isPhysical()) {
+      if (!SrcReg.isPhysical()) {
+        report("Copy with lanemask Instruction uses virtual register", &SrcOp,
+               1);
+      }
+      if (!DstReg.isPhysical()) {
+        report("Copy with lanemask Instruction uses virtual register", &DstOp,
+               0);
+      }
+      break;
+    }
+
+    if (LaneMask.none())
```

Comment:
I'm not sure we should be enforcing this not-none and not-all conditions, but let's go with it for now.